src/vue/backend/src/components/APIKey.vue

```diff
@@ -0,0 +1,103 @@
+<template>
+    <idx-form-group class="idx-wp-needs-validation" :class="{'idx-wp-was-validated': success|error}" novalidate>
+        <idx-form-label for="APIKey" class="idx-wp-visually-hidden">API Key</idx-form-label>
+        <idx-form-input
+            type="text"
+            id="APIKey"
+            :placeholder="placeholder"
+            :class="{
+                'is-invalid': error,
+                'is-valid': success,
+                'is-loading': loading
+            }"
+            :invalid="error"
+            :valid="success"
+            required
+        />
+        <idx-block className="invalid-feedback" v-if="error">
+            We couldn't find an account with the provided API key
+        </idx-block>
+        <idx-block className="idx-wp-spinner-border" role="status" v-if="loading">
+            <idx-block tag="span" className="idx-wp-visually-hidden">Loading...</idx-block>
+        </idx-block>
+    </idx-form-group>
+</template>
+
+<script>
+export default {
+    name: 'APIKey',
+    props: {
+        placeholder: {
+            type: String,
+            default: 'Enter Your API Key'
+        },
+        error: {
+            type: Boolean,
+            default: true
+        },
+        loading: {
+            type: Boolean,
+            default: false
+        },
+        success: {
+            type: Boolean,
+            default: false
+        }
+    }
+}
+</script>
+
+<style scoped lang="scss">
+    @import '~@idxbrokerllc/idxstrap/dist/styles/globalVariables.scss';
```

Comment:
We'll want these in the main.js file, this is fine for now but we will remove it later